akka-javasdk/src/main/scala/akka/javasdk/impl/SdkRunner.scala

```diff
@@ -673,12 +673,14 @@ private final class Sdk(
           new AgentDescriptor(componentId, clz.getName, instanceFactory)
 
         agentRegistryInfo :+=
-          AgentRegistryImpl.AgentDetails(
-            componentId,
-            agentDescription.name,
-            agentDescription.description,
-            agentDescription.role,
-            agentClass)
+          (agentDescription match {
+            case Some(desc) =>
+              AgentRegistryImpl.AgentDetails(componentId, desc.name, desc.description, desc.role, agentClass)
+            case None =>
+              // defaults if AgentDescription is not defined
+              val name = componentId.replace('-', ' ').replace('_', ' ')
+              AgentRegistryImpl.AgentDetails(componentId, name, name, "", agentClass)
```

Comment:
Wonder if just reusing the component id for the name as is, and empty description, would be better defaults. Rather than expecting particular naming patterns.

Reply:
I agree that we could simply use the component id. 
But I don't think it's a good idea to have an empty description. 

An agent without a description can't be selected by a planner or selector. It's only useful for static agents calls where we know exactly who we need to call.

Do we think that it's such a common use case to justify making the description optional? 

I would consider it more as a edge case. Therefore I would rather make the description mandatory and make sure users don't forget it. 

The name could stay optional and default to component id.

Reply:
Ok, I came late with my comment. This PR was open on a tab and I didn't see it was already merged.

Reply:
Not sure we are talking about the same thing.

The purpose of this PR was to make the annotation `@AgentDescription` optional, because I saw that as much boilerplate without any value when AgentRegistry (planner) isn't used.

Peter's comment was about the default value that was filled in when the `@AgentDescription` isn't defined.

If `@AgentDescription` is defined, the `description` field is mandatory and can't be blank.